vnext/Shared/JSI/NapiJsiRuntime.h

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+// JSI
+#include <js_native_ext_api.h>
+#include <jsi/jsi.h>
+
+// Standard Library
+#include <memory>
+
+namespace Microsoft::JSI {
+
+///
+// NapiJsiRuntime factory function.
+// TODO: Rename as MakeNapiJsiRuntime once code is dropped from V8-JSI.
+///
+std::unique_ptr<facebook::jsi::Runtime> __cdecl MakeNapiJsiRuntime2(napi_env env) noexcept;
```

Comment:
>__cdecl [](http://example.com/codeflow?start=40&length=7)

(Nit) Probably no longer needed since we don't export it

Reply:
Added the symbol to the DLL boundary to make it accessible for integration tests.
Keeping the explicit calling convention for now.

Reply:
Since we do not want this function to be in the ABI boundary, it would be better if tests compile this code as the consumers of the DLL must do.